user/service.go

```diff
@@ -12,6 +12,10 @@ type UserEntity struct {
 // Set time.Now outside of method for testing
 var timeNow = time.Now
 
+func NewEntity(dataSource UserDataSource) UserEntity {
```

Comment:
What is this for?

Reply:
UserEntity's `dataSource` is private field, which means it is not possible to set it outside of this package. 
`NewEntity` is a factory function which enables consumers of this package to create `UserEntity`.